src/services/programService/httpClient.js

```diff
@@ -46,6 +53,9 @@ export const listDataCenters = async (jwt) => {
 		},
 	})
 		.then(restErrorResponseHandler)
-		.then((response) => response.json());
+		.then((response) => response.json())
+		.then((response) => {
+			return shortName ? getDataCenterByShortName(shortName, response) : response;
+		});
 	return response;
 };
```

Comment:
add a condition of shortName being used or not, and return corresponding response to the resolver.